lib/rubocop/cop/lint/empty_conditional_body.rb

```diff
@@ -107,14 +108,36 @@ def correct_other_branches(corrector, node)
           end
         end
 
+        def correctable_condition?(node)
+          node.if? || node.unless? || node.elsif?
+        end
+
+        def empty_else_branch?(node)
+          node.else_branch.if_type? && !node.else_branch.body
+        end
+
         def branch_range(node)
           if node.loc.else
             node.source_range.with(end_pos: node.loc.else.begin_pos - 1)
+          elsif all_branches_body_missing?(node)
+            node.source_range.with(end_pos: parent_if_node(node).loc.end.end_pos)
```

Comment:
Can you use `node.ancestors.detect(&:if?)` instead of the new `parent_if_node` method?

```suggestion
            if_node = node.ancestors.detect(&:if?)

            node.source_range.with(end_pos: if_node.loc.end.end_pos)
```

`Metrics/AbcSize`'s warning will probably be displayed, but I think that `branch_range` method can be enclosed with `# rubocop:disable Metrics/AbcSize` and `# rubocop:enable Metrics/AbcSize`.

Reply:
Indeed it is. Use `node.ancestors.detect(&:if?)`. Thank you!